src/main/java/co/cask/hydrator/plugin/DateTransform.java

```diff
@@ -99,7 +99,8 @@ public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter)
       String sourceField = sourceFields.get(i);
       String targetField = targetFields.get(i);
 
-      if (input.getSchema().getField(sourceField) == null) {
+      if (input.getSchema().getField(sourceField) == null ||
```

Comment:
Shouldn't it fail when the fields present in source fields config is not present in the input schema. We should check it while configuring a pipeline, somewhere here
https://github.com/hydrator/date-transform/blob/a0a64f1878ed5b75fc3212e3b1457821f3f41371/src/main/java/co/cask/hydrator/plugin/DateTransform.java#L255

```
if (inputSchema.getField(field) == null) {
throw exception;
};
```